src/Utilities/Compiler/DiagnosticDescriptorHelper.cs

```diff
@@ -60,6 +60,7 @@ internal static partial class DiagnosticDescriptorHelper
                     RuleLevel.IdeHidden_BulkConfigurable => (DiagnosticSeverity.Hidden, true),
                     RuleLevel.Disabled => (DiagnosticSeverity.Warning, false),
                     RuleLevel.CandidateForRemoval => (DiagnosticSeverity.Warning, false),
+                    RuleLevel.BuildError => (DiagnosticSeverity.Error, true),
```

Comment:
`DiagnosticDescriptorCreationAnalyzer` may need to get updated to handle this new RuleLevel.

Reply:
It could be updated yea, but we should do it in a separate PR once this goes in